pkg/login/social/generic_oauth.go

```diff
@@ -260,24 +288,34 @@ func (s *SocialGenericOAuth) extractToken(data *UserInfoJson, token *oauth2.Toke
 	return true
 }
 
-func (s *SocialGenericOAuth) extractEmail(data *UserInfoJson) string {
+func (s *SocialGenericOAuth) extractEmail(data *UserInfoJson, userInfoResp []byte) (string, error) {
 	if data.Email != "" {
-		return data.Email
+		return data.Email, nil
 	}
 
+	email, err := s.searchJSONForEmail(userInfoResp)
+	if err != nil {
+		s.log.Debug("Failed to search user info JSON response for e-mail", "err", err.Error())
+	} else {
+		return email, nil
+	}
+	s.log.Debug("No e-mail address found when searching user info JSON response")
```

Comment:
This line isn't needed, as we'll always log the actual error above.  The flow would be logically the same but may be easier to read if it were written `if err == nil {`, the same is true on lines 311-314 below.